doc/tutorials/objdetect/aruco_detection/aruco_detection.markdown

```diff
@@ -0,0 +1,802 @@
+-c=/path_to_opencv/opencv/samples/cpp/tutorial_code/objectDetection/tutorial_camera_params.yml
+@endcode
+Parameters for `detect_markers.cpp`:
+@snippet samples/cpp/tutorial_code/objectDetection/detect_markers.cpp aruco_detect_markers_keys
+
+@note To work with examples from the tutorial, you can use camera parameters from `tutorial_camera_params.yml`.
+An example of use in `detect_markers.cpp`.
+
+
+
+Selecting a dictionary
+------
+
+The aruco module provides the `Dictionary` class to represent a dictionary of markers.
+
+In addition to the marker size and the number of markers in the dictionary, there is another important parameter of the dictionary -
+the inter-marker distance. The inter-marker distance is the minimum distance between dictionary markers
+that determines the dictionary's ability to detect and correct errors.
+
+In general, smaller dictionary sizes and larger marker sizes increase the inter-marker distance and
+vice versa. However, the detection of markers with larger sizes is more difficult due to the higher
+number of bits that need to be extracted from the image.
+
+For instance, if you need only 10 markers in your application, it is better to use a dictionary composed only of those 10 markers than using a dictionary composed of 1000 markers. The reason is that
+the dictionary composed of 10 markers will have a higher inter-marker distance and, thus, it will be
+more robust to errors.
+
+As a consequence, the aruco module includes several ways to select your dictionary of markers, so that
+you can increase your system robustness:
+
+### Predefined dictionaries
+
+This is the easiest way to select a dictionary. The aruco module includes a set of predefined
+dictionaries in a variety of marker sizes and number of markers. For instance:
+
+@code{.cpp}
+cv::aruco::Dictionary dictionary = cv::aruco::getPredefinedDictionary(cv::aruco::DICT_6X6_250);
+@endcode
+
+`cv::aruco::DICT_6X6_250` is an example of predefined dictionary of markers with 6x6 bits and a total of 250
+markers.
+
+From all the provided dictionaries, it is recommended to choose the smallest one that fits your application.
+For instance, if you need 200 markers of 6x6 bits, it is better to use `cv::aruco::DICT_6X6_250` than `cv::aruco::DICT_6X6_1000`.
+The smaller the dictionary, the higher the inter-marker distance.
+
+The list of available predefined dictionaries can be found in the documentation for the `PredefinedDictionaryType` enum.
+
+### Automatic dictionary generation
+
+A dictionary can be generated automatically to adjust the desired number of markers and bits to optimize the inter-marker distance:
+
+@code{.cpp}
+cv::aruco::Dictionary dictionary = cv::aruco::extendDictionary(36, 5);
+@endcode
+
+This will generate a customized dictionary composed of 36 markers of 5x5 bits. The process can take several
+seconds, depending on the parameters (it is slower for larger dictionaries and higher numbers of bits).
+
+### Manual dictionary generation
```

Comment:
not generation, but definition.

Reply:
fixed:

`### Manual dictionary definition`